src/aiokatcp/sensor.py

```diff
@@ -714,3 +748,127 @@ def copy(self) -> Dict[str, Sensor]:
     values.__doc__ = dict.values.__doc__
     items.__doc__ = dict.items.__doc__
     copy.__doc__ = dict.copy.__doc__
+
+
+class AggregateSensor(Sensor[_T], metaclass=ABCMeta):
+    """A Sensor with its reading determined by several other Sensors.
+
+    This is an abstract class: the user must implement :meth:`update_aggregate`.
+    This method is called whenever the target :class:`SensorSet` has a sensor
+    added, removed, or one of the sensors changes its reading. The user-defined
+    :meth:`update_aggregate` returns the new reading for the AggregateSensor.
+
+    Parameters are all as per :class:`Sensor`, with the exception of
+    `target`, which is the :class:`SensorSet` from which the aggregated sensor
+    will determine its own reading.
+
+    Attributes
+    ----------
+    target : SensorSet
+        The set of sensors which will determine the reading of the aggregate
+        one. The aggregate sensor may be included in the set (e.g. in
+        `self.sensors` of a server) but it will not affect its own value.
+    """
+
+    def __init__(
+        self,
+        target: SensorSet,
+        sensor_type: Type[_T],
+        name: str,
+        description: str = "",
+        units: str = "",
+        *,
+        auto_strategy: Optional["SensorSampler.Strategy"] = None,
+        auto_strategy_parameters: Iterable[Any] = (),
+    ) -> None:
+        super().__init__(
+            sensor_type=sensor_type,
+            name=name,
+            description=description,
+            units=units,
+            auto_strategy=auto_strategy,
+            auto_strategy_parameters=auto_strategy_parameters,
+        )
+        self.target = target
+        for sensor in self.target.values():
+            if self.filter_aggregate(sensor):
+                sensor.attach(self._update_aggregate)
+
+        self.target.add_add_callback(self._sensor_added)
+        self.target.add_remove_callback(self._sensor_removed)
+        self._update_aggregate(None, None, None)  # type: ignore
```

Comment:
Why the `type: ignore`?

Reply:
Might have been a leftover while I was still fighting with mypy. Removed.